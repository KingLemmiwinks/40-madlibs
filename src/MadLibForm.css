#mad-libs-form input {
    display: block;
    margin: 10px auto;
    padding: 5px;
    border-radius: 8px;
}