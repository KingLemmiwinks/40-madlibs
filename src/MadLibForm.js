import React from "react";
import "./MadLibForm.css";

const MadLibForm = ({ userWords, wordsFormHandleChange, submitUserWords }) => {
  return (
    <form id="mad-libs-form" onSubmit={submitUserWords}>
      {Object.keys(userWords).map((wordType, idx) => (
        <input
          key={idx}
          type="text"
          placeholder={wordType}
          name={wordType}
          value={userWords.wordType}
          onChange={wordsFormHandleChange}
          required
        />
      ))}
      <button>Show Story!</button>
    </form>
  );
};

export default MadLibForm;