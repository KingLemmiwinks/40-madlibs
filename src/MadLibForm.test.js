import React from 'react';
import { render } from '@testing-library/react';
import MadLibForm from './MadLibForm';

test('it will render without crashing', () => {
    render(<MadLibForm />)
})