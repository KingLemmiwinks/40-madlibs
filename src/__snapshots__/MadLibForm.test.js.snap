// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MadLibForm Tests matches snapshot 1`] = `
<DocumentFragment>
  <form
    id="mad-libs-form"
  >
    <input
      name="0"
      placeholder="0"
      required=""
      type="text"
      value=""
    />
    <input
      name="1"
      placeholder="1"
      required=""
      type="text"
      value=""
    />
    <input
      name="2"
      placeholder="2"
      required=""
      type="text"
      value=""
    />
    <input
      name="3"
      placeholder="3"
      required=""
      type="text"
      value=""
    />
    <button>
      Show Story!
    </button>
  </form>
</DocumentFragment>
`;
